lib/input_sanitizer/v2/nested_sanitizer_factory.rb

```diff
@@ -8,10 +8,30 @@ def valid?
       true
     end
   end
+  
+  class HashExpected
+    def initialize(value)
+      @value = value
+    end
+    
+    def valid?
+      false
+    end
+    
+    def cleaned
+      nil
+    end
+    
+    def errors
+      [InputSanitizer::TypeMismatchError.new(@value, :hash)]
+    end
+  end
 
   def self.for(nested_sanitizer_klass, value, options)
     if value.nil? && options[:allow_nil] && !options[:collection]
       NilAllowed.new
+    elsif !value.is_a?(Hash)
```

Comment:
Shouldn't there also be a check for Hash inside the array? 🤔 By that I mean an additional condition with `options[:collection]` and some kind of `value.all?`, if that option even applies to a nested field type

Reply:
I'm in the wrong, each value provided to field with `collection => true` is already enumerated into this `nested_sanitizer_factory`, so there would never be value here of type array